snorkel/labeling/apply/lf_applier.py

```diff
@@ -0,0 +1,44 @@
+from itertools import chain
+from typing import Any, Iterable, List, Tuple
+
+import scipy.sparse as sparse
+from tqdm import tqdm
+
+from snorkel.labeling.lf import LabelingFunction
+from snorkel.types import Example
```

Comment:
I don't like the name `Example`.  What about `DataPoint`

Reply:
Open to suggestions but "example" is weird to me

Reply:
allennlp uses `Instance`, though `DataPoint` is probably the most clear

Reply:
Either way we should be consistent in our verbiage across both code and text (whenever possible).  We say "LFs to label _training data_", but "data" is plural; "datum" sounds pretentious... so that led me to "data point"

Reply:
Just to clarify, Example = "training example". But I'm open to whatever. Any objections to DataPoint?

Reply:
DataPoint sgtm. Just checking: Example can refer to both examples used for training and testing, right?

Reply:
Actually why not just use Instance if it's already in use by a popular package?

Reply:
`DataPoint` maintains a clearer mapping with our overall explanation of how Snorkel works, e.g. in tutorials, etc.